### PR TITLE
Document JSON value constructors

### PR DESCRIPTION
**Pull request by kasiafi**
Documentation for https://github.com/trinodb/trino/pull/12049.

### CODE REVIEW

File: docs/src/main/sphinx/functions/json.rst [hunk 1]
```diff
@@ -17,6 +17,9 @@ Trino supports three functions for querying JSON data:
 is based on the same mechanism of exploring and processing JSON input using
 JSON path.
 
+Trino also supports two functions for generating JSON data:
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:20]: either make it a sentence without a : .. or keep the : but then actually enumerate them in a list. In this case I would probably just replace : with -

File: docs/src/main/sphinx/functions/json.rst [hunk 2]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+.. code-block:: text
+
+    SELECT json_array(
+                      '[  "text"  ] ' FORMAT JSON,
+                      X'5B0035005D00' FORMAT JSON ENCODING UTF16
+                     )
+    --> '[["text"],[5]]'
+
+You can also nest other json-returning functions. In that case, the ``FORMAT``
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1209]: JSON-returning

File: docs/src/main/sphinx/functions/json.rst [hunk 3]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+option is implicit:
+
+.. code-block:: text
+
+    SELECT json_array(json_query('{"key" : [  "value"  ]}', 'lax $.key'))
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1214]: should we break the nested json_query in separate line below with indentation?

File: docs/src/main/sphinx/functions/json.rst [hunk 4]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+    SELECT json_array() --> '[]'
+
+Null handling
+^^^^^^^^^^^^^
+
+If a value passed for an array element is ``null``, it is treated according to
+the specified null treatment option. If ``ABSENT ON NULL`` is specified, the
+null element is omitted in the result. If ``NULL ON NULL`` is specified, JSON
+``null`` is added to the result. ``ABSENT ON NULL`` is default.
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1239]: is the default.

or 

is the default configuration.

File: docs/src/main/sphinx/functions/json.rst [hunk 5]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+.. code-block:: text
+
+    SELECT json_array(true, null, 1)                --> '[true,1]'
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1243]: should we put result in next line and add empty line after like in other snippets below .. it make it nicer to read and to cut and paste from..

File: docs/src/main/sphinx/functions/json.rst [hunk 6]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+    SELECT json_array(true, null, 1 ABSENT ON NULL) --> '[true,1]'
+    SELECT json_array(true, null, 1 NULL ON NULL)   --> '[true,null,1]'
+
+Returned type
+^^^^^^^^^^^^^
+
+The SQL standard imposes that there is no dedicated data type to represent JSON
+data in SQL. Instead, JSON data is represented as character or binary strings.
+By default, the ``json_array`` function returns varchar containing textual
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1252]: containing the textual

File: docs/src/main/sphinx/functions/json.rst [hunk 7]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+    SELECT json_array(true, 1 RETURNING VARBINARY FORMAT JSON ENCODING UTF32)
+    --> X'5b 00 00 00 74 00 00 00 72 00 00 00 75 00 00 00 65 00 00 00 2c 00 00 00 31 00 00 00 5d 00 00 00'
+
+.. _json_object:
+
+json_object
+-----------
+
+The ``json_object`` function Creates a JSON object containing given key-value pairs.
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1282]: creates

File: docs/src/main/sphinx/functions/json.rst [hunk 8]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+Argument types
+^^^^^^^^^^^^^^
+
+The keys can be arbitrary expressions. They must be of character string type.
+Each key is converted into a JSON text item, and it becomes a key in the
+created JSON object. Keys must not be null.
+
+The values can be arbitrary expressions. Each passed value is converted
+into a JSON item according to its type and optional ``FORMAT`` and ``ENCODING``
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1320]: type, and

File: docs/src/main/sphinx/functions/json.rst [hunk 9]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+You can pass SQL values of types boolean, numeric, and character string. They
+are converted to corresponding JSON literals:
+
+.. code-block:: text
+
+    SELECT json_object('x' : true, 'y' : 12e-1, 'z' : 'text')
+    --> '{"x":true,"y":1.2,"z":"text"}'
+
+Additionally to SQL values, you can pass JSON values. They are character or
+binary strings with specified format and optional encoding:
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1332]: with a specified

File: docs/src/main/sphinx/functions/json.rst [hunk 10]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+.. code-block:: text
+
+    SELECT json_object(
+                       'x' : '[  "text"  ] ' FORMAT JSON,
+                       'y' : X'5B0035005D00' FORMAT JSON ENCODING UTF16
+                      )
+    --> '{"x":["text"],"y":[5]}'
+
+You can also nest other json-returning functions. In that case, the ``FORMAT``
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1342]: JSON-returning

File: docs/src/main/sphinx/functions/json.rst [hunk 11]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+The values passed for JSON object keys must not be null. It is allowed to pass
+``null`` for JSON object values. A null value is treated according to the
+specified null treatment option. If ``NULL ON NULL`` is specified, a JSON
+object entry with ``null`` value is added to the result. If ``ABSENT ON NULL``
+is specified, the entry is omitted in the result. ``NULL ON NULL`` is default.
+
+.. code-block:: text
+
+    SELECT json_object('x' : null, 'y' : 1)                --> '{"x":null,"y":1}'
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1377]: Again .. maybe break up

File: docs/src/main/sphinx/functions/json.rst [hunk 12]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+
+If ``WITH UNIQUE KEYS`` is specified, a duplicate key results in a query
+failure:
+
+.. code-block:: text
+
+    SELECT json_object('x' : null, 'x' : 1 WITH UNIQUE KEYS)
+    --> failure: "duplicate key passed to JSON_OBJECT function"
+
+
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1395]: remove empty line

File: docs/src/main/sphinx/functions/json.rst [hunk 13]
```diff
@@ -1163,6 +1166,269 @@ id         child
...
+Note that this option is not supported if any of the arguments has
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/json.rst:1396]: has a

**Review by mosabua — CHANGES_REQUESTED** [review]
Great job. Just minor things. 

One question I have .. you chose to not use sql for the code blocks but text instead .. and also did not rely on the the :: before the snippets. Is that on purpose because it does not work somehow for you? If not .. we might want to change them all

**Review by losipiuk — APPROVED** [review]

**Pull request merged**